client/src/components/ProductOverview/ProductImage.jsx

```diff
@@ -32,65 +34,109 @@ const ProductImage = (props) => {
     }
   };
 
+  const handleImageclick = e => {
+    let current = e.target;
+    let elements = [current];
+
+    while (current.parentElement) {
+      elements.push(current.parentElement.id);
+      current = current.parentElement;
+    }
+
+    if (elements.includes('image-list')) {
+      return;
+    }
+    if (elements.includes('image-arrows')) {
+      return;
+    }
+    if (elements.includes('fullscreen-toggle')) {
+      return;
+    }
+
+    document.body.style.height = 'auto';
+    document.body.style.overflowY = 'scroll';
+
+    if (!props.fullscreen) {
+      props.toggleFullScreen();
+    } else {
+      setZoomed(!zoomed);
+
+      if (!zoomed) {
+        document.body.style.height = '100vh';
+        document.body.style.overflowY = 'hidden';
+
+        const img = document.querySelector('#product-image');
+
+        img.addEventListener('mousemove', e => {
+          let width = parseInt(window.getComputedStyle(document.querySelector('#product-image')).getPropertyValue('width'));
+          let height = parseInt(window.getComputedStyle(document.querySelector('#product-image')).getPropertyValue('height'));
+
+          img.style.setProperty('--x', width * 2.5 - e.pageX + 'px');
+          img.style.setProperty('--y', height * 2.5 - e.pageY + 'px');
+        });
+      }
+    }
+  };
 
   return (
-    <div id="product-image" className="bg-image" style={{ backgroundImage: props.images[currentImageIndex]?.url ? `url(${props.images[currentImageIndex]?.url})` : 'none' }}>
-      <div id="image-list">
-        <div className="image-alt-scroll up" style={{ color: firstImageInd === 0 ? 'transparent' : '#454545' }}>
-          <AiOutlineUp
-            className="clickable"
-            onClick={() => scroll('up')}
-          />
+    <div
+      id='product-image'
+      className={`bg-image ${props.fullscreen ? 'fullscreen' : ''} ${zoomed ? 'zoomed' : ''}`}
```

Comment:
Next time I start a project, I am going to deconstruct props everytime. I wrote 'props' way too many times this round.

Reply:
good call!